sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs

```diff
@@ -1720,17 +1744,20 @@ public virtual Response DeleteEventRoute(string eventRouteId, CancellationToken
         /// <example>
         /// <code snippet="Snippet:DigitalTwinsSamplePublishTelemetry">
         /// // construct your json telemetry payload by hand.
-        /// await client.PublishTelemetryAsync(twinId, &quot;{\&quot;Telemetry1\&quot;: 5}&quot;);
+        /// await client.PublishTelemetryAsync(twinId, Guid.NewGuid().ToString(), &quot;{\&quot;Telemetry1\&quot;: 5}&quot;);
         /// Console.WriteLine($&quot;Published telemetry message to twin &apos;{twinId}&apos;.&quot;);
         /// </code>
         /// </example>
-        public virtual Task<Response> PublishTelemetryAsync(string digitalTwinId, string payload, TelemetryOptions options = default, CancellationToken cancellationToken = default)
+        public virtual Task<Response> PublishTelemetryAsync(string digitalTwinId, string messageId, string payload, DigitalTwinsSendTelemetryOptions options = null, CancellationToken cancellationToken = default)
```

Comment:
In the latest swagger, messageId is actually listed as mandatory, even if it is just a guid. Because of that, I'm opting to move it out from the options object and made it a mandatory parameter like digital twin id and payload. This feels debatable to me though, so I'd be curious to hear everyone's thoughts on this

Reply:
I think it could still be optional with a default value of a guid we generate, but it depends on how the service team feels. Do they want the user to set one by putting in their face (required parameter)?

Reply:
I personally think we should generate a random guid on their behalf and let them override it if they are interested in doing so.

Reply:
We can certainly generate a GUID on the users behalf if they pass in a null value. I've already added that implementation so that it is consistent with the preview SDK we released. My bigger question is if this messageId should live in the options bundle or not. Technically it is not optional, but it can be arbitrary. It certainly feels like an option though.

Reply:
Which is why I think you should ask the service team how they want it to appear in the API.